frontend/src/components/QuestionInput/QuestionInput.test.tsx

```diff
@@ -0,0 +1,162 @@
+import { ACCEPTED_FILE_TYPES } from '../../custom/fileUploadUtils'
+import { QuestionInput } from './QuestionInput'
+
+import { axe, toHaveNoViolations } from 'jest-axe'
+expect.extend(toHaveNoViolations)
+
+async function uploadFile(file: File) {
+  const fileInput = screen.getByLabelText('Upload file')
+
+  await fireEvent.change(fileInput, { target: { files: [file] } })
+}
+
+function inputChatMessage(message?: string) {
+  const textInputField = screen.getByLabelText('Type a question')
+
+  fireEvent.change(textInputField, {
+    target: { value: message ?? 'This is my message' }
+  })
+}
+
+function submitChatMessage() {
+  const submitButton = screen.getByLabelText('Ask question button')
+
+  fireEvent.click(submitButton)
+}
+
+const createMockFile = (extension: string, fileType: string): File => {
+  const blob = new Blob(['hello'], { type: fileType })
+  const file = new File([blob], `default.${extension}`, { type: fileType })
+
+  return file
+}
+
+describe('Test uploading files', () => {
+  afterEach(() => {
+    jest.clearAllMocks()
+  })
+
+  it('uploads PDFs without errors', async () => {
+    const uploadedFile = createMockFile('pdf', ACCEPTED_FILE_TYPES.PDF)
+
+    const { container } = render(
+      <QuestionInput
+        onSend={jest.fn()}
+        disabled={false}
+        placeholder={'placeholder'}
+        conversationId={undefined}
+        clearOnSend={false}
+      />
+    )
+
+    await act(async () => {
+      uploadFile(uploadedFile)
+      inputChatMessage()
+      submitChatMessage()
+    })
+
+    expect(await axe(container)).toHaveNoViolations();
+  })
+
+  it('uploads CSVs without errors', async () => {
+    const uploadedFile = createMockFile('csv', ACCEPTED_FILE_TYPES.CSV)
+
+    const { container } = render(
+      <QuestionInput
+        onSend={() => {}}
+        disabled={false}
+        placeholder={'placeholder'}
+        conversationId={undefined}
+        clearOnSend={false}
+      />
+    )
+
+    await act(async () => {
+      uploadFile(uploadedFile)
+      inputChatMessage()
+      submitChatMessage()
+    })
+
+    expect(await axe(container)).toHaveNoViolations();
+  })
+
+  it('uploads .docx files without errors', async () => {
+    const uploadedFile = createMockFile('.docx', ACCEPTED_FILE_TYPES.DOCX)
+
+    const { container } = render(
+      <QuestionInput
+        onSend={() => {}}
+        disabled={false}
+        placeholder={'placeholder'}
+        conversationId={undefined}
+        clearOnSend={false}
+      />
+    )
+
+    await act(async () => {
+      uploadFile(uploadedFile)
+      inputChatMessage()
+      submitChatMessage()
+    })
+
+    expect(await axe(container)).toHaveNoViolations();
+  })
+
+  it.each([
+    ['jpg', ACCEPTED_FILE_TYPES.JPEG],
+    ['png', ACCEPTED_FILE_TYPES.PNG],
+    ['gif', ACCEPTED_FILE_TYPES.GIF],
+    ['bmp', ACCEPTED_FILE_TYPES.BMP],
+    ['tiff', ACCEPTED_FILE_TYPES.TIFF]
+  ])(
```

Comment:
[dust] - i like the `it.each()` and i wonder if the other file types should be put in here too, since it looks like the contents of the PDF, CSV, and DOCX are also the same. So it could be one test for all file types?